diff --git a/app/code/Cloras/DDI/Model/Shipto.php b/app/code/Cloras/DDI/Model/Shipto.php
index 0ed6235..6b9cf7c 100755
--- a/app/code/Cloras/DDI/Model/Shipto.php
+++ b/app/code/Cloras/DDI/Model/Shipto.php
@@ -58,7 +58,28 @@ class Shipto implements ShiptoInterface
                             if ($ddiShipNumber == $erpAddress['shipNumber'] && $erpAccountNumber == $accountNumber) {
                                 $firstName = $customerAddress->getFirstname();
                                 $lastName = $customerAddress->getLastname();
-                                if (isset($erpAddress['shipName']) && $erpAddress['shipName'] != '') {
+                                /*if (isset($erpAddress['shipName']) && $erpAddress['shipName'] != '') {
+                                    $shipCompanyName = explode(' ', $erpAddress['shipName']);
+                                    if (count($shipCompanyName) > 1) {
+                                        $firstName = $shipCompanyName[0];
+                                        unset($shipCompanyName[0]);
+                                        $lastName = implode(' ', $shipCompanyName);
+                                    } else {
+                                        $firstName = $shipCompanyName[0];
+                                        $lastName = $shipCompanyName[0];
+                                    }
+                                }*/
+                                if (isset($erpAddress['shipCompanyName']) && $erpAddress['shipCompanyName'] != '') {
+                                    $shipCompanyName = explode(' ', $erpAddress['shipCompanyName']);
+                                    if (count($shipCompanyName) > 1) {
+                                        $firstName = $shipCompanyName[0];
+                                        unset($shipCompanyName[0]);
+                                        $lastName = implode(' ', $shipCompanyName);
+                                    } else {
+                                        $firstName = $shipCompanyName[0];
+                                        $lastName = $shipCompanyName[0];
+                                    }
+                                } elseif (isset($erpAddress['shipName']) && $erpAddress['shipName'] != '' && isset($erpAddress['shipCompanyName']) && $erpAddress['shipCompanyName'] == '') {
                                     $shipCompanyName = explode(' ', $erpAddress['shipName']);
                                     if (count($shipCompanyName) > 1) {
                                         $firstName = $shipCompanyName[0];
@@ -69,10 +90,10 @@ class Shipto implements ShiptoInterface
                                         $lastName = $shipCompanyName[0];
                                     }
                                 }
-                                if (isset($erpAddress['shipName']) && $erpAddress['shipName'] == '') {
+                                /*if (isset($erpAddress['shipName']) && $erpAddress['shipName'] == '') {
                                     $firstName = "Attn:";
                                     $lastName = "Not Provided";
-                                }
+                                }*/
 
                                 $region = $this->regionFactory->create();
                                 $countryId = 'US';
@@ -157,7 +178,17 @@ class Shipto implements ShiptoInterface
         $customerId = $customerRepository->getId();
         $firstName = $customerRepository->getFirstname();
         $lastName = $customerRepository->getLastname();
-        if (isset($erpAddress['shipName']) && $erpAddress['shipName'] != '') {
+        if (isset($erpAddress['shipCompanyName']) && $erpAddress['shipCompanyName'] != '') {
+            $shipCompanyName = explode(' ', $erpAddress['shipCompanyName']);
+            if (count($shipCompanyName) > 1) {
+                $firstName = $shipCompanyName[0];
+                unset($shipCompanyName[0]);
+                $lastName = implode(' ', $shipCompanyName);
+            } else {
+                $firstName = $shipCompanyName[0];
+                $lastName = $shipCompanyName[0];
+            }
+        } elseif (isset($erpAddress['shipName']) && $erpAddress['shipName'] != '' && isset($erpAddress['shipCompanyName']) && $erpAddress['shipCompanyName'] == '') {
             $shipCompanyName = explode(' ', $erpAddress['shipName']);
             if (count($shipCompanyName) > 1) {
                 $firstName = $shipCompanyName[0];
@@ -168,10 +199,10 @@ class Shipto implements ShiptoInterface
                 $lastName = $shipCompanyName[0];
             }
         }
-        if (isset($erpAddress['shipName']) && $erpAddress['shipName'] == '') {
+        /*if (isset($erpAddress['shipName']) && $erpAddress['shipName'] == '') {
             $firstName = "Attn:";
             $lastName = "Not Provided";
-        }
+        }*/
         $countryData = $this->getCountryData();
 
         $region = $this->regionFactory->create();
