diff --git a/app/code/Cloras/DDI/Observer/CustomerloginObserver.php b/app/code/Cloras/DDI/Observer/CustomerloginObserver.php
index 01a4db2..eb8ec13 100755
--- a/app/code/Cloras/DDI/Observer/CustomerloginObserver.php
+++ b/app/code/Cloras/DDI/Observer/CustomerloginObserver.php
@@ -76,6 +76,18 @@ class CustomerloginObserver implements ObserverInterface
                 if ($this->extensionHelper->getIsLogger() && $this->isB2B()) {
                     $this->logger->info('Ecomm user details added in session');
                 }
+            } else {
+                /* this is used for login as customer fix */
+                list($status, $integrationData) = $this->ddiHelper->isServiceEnabled('validate_user');
+                if ($status) {
+                    $users = $this->ddiHelper->validateEcommUser($integrationData, $customerSession->getCustomer()->getEmail());
+                    if ($users && isset($users['isValid']) && $users['isValid'] == 'yes') {
+                        if (isset($users[0]['token'])) {
+                            $this->coreCustomerSession->setEcommtoken(['ecomm_token' => $users[0]['token']]);
+                        }
+                        $this->coreCustomerSession->setCustomData($users[0]);
+                    }
+                }
             }
         }
     }
